resources-generator/src/main/kotlin/dev/icerock/gradle/utils/LifecycleAwareCodeLauncher.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2023 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.gradle.utils
+
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.getPlugin
+import org.jetbrains.kotlin.gradle.plugin.KotlinMultiplatformPluginWrapper
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+import org.slf4j.LoggerFactory
+import java.lang.reflect.Method
+
+/**
+ * Launches [block] when populating of all [KotlinSourceSet.dependsOn] SourceSets is complete.
+ */
+internal fun Project.launchWhenEdgeRefinementFinalized(block: () -> Unit) {
+    val version = project.plugins.getPlugin(KotlinMultiplatformPluginWrapper::class).pluginVersion
+        .split(".")
+        .take(2)
+        .map(String::toInt)
+
+    @Suppress("MagicNumber")
+    when {
+        version[0] >= 1 && version[1] >= 9 -> {
```

Comment:
`version[0] >= 1 && version[1] >= 9` returns false for 2.4, for example, we need something like `version[0] > 1 || version[0] = 1 && version[1] >= 9`

Reply:
good point, thank you!